pkg/operator/controller/ingress/deployment.go

```diff
@@ -646,6 +647,44 @@ func (r *reconciler) updateRouterDeployment(current, desired *appsv1.Deployment)
 		return nil
 	}
 
+	// It may be impossible to transition from the current state to the
+	// desired state if the current state has an affinity policy with an
+	// anti-affinity rule that prohibits colocation of pods belonging to the
```

Comment:
Does "anti-affinity rule" cover if the rule is type `nodeAffinity` or `podAffinity`?

Reply:
I think so.  In this case, it is a pod anti-affinity rule.  I can add the word "pod" to the comment.

Reply:
Done.